pecos/core/utils/clustering.hpp

```diff
@@ -102,6 +153,13 @@ struct Tree {
         right.set(middle, root.end);
     }
 
+    void sample_elements(Node& root, rng_t& rng, float cur_sample_rate) {
+        rng.shuffle(elements.begin() + root.start, elements.begin() + root.end);
+        size_t n_sp_elements = size_t(cur_sample_rate * root.size());
+        n_sp_elements = std::min(std::max(n_sp_elements, size_t(2)), root.size()); // clamp the value sampled elements
```

Comment:
hard-coded `size_t(2)` seems arbitrary. Better to define it to be a CONSTANT variable in the struct of `ClusteringSampler` or `ClusteringSamplerParam`.

Reply:
We throw the invalid arguments at ClusteringSamplerParam now.